lib/fixed-points.nix

```diff
@@ -107,7 +107,7 @@ rec {
   # Same as `makeExtensible` but the name of the extending attribute is
   # customized.
   makeExtensibleWithCustomName = extenderName: rattrs:
-    fix' rattrs // {
+    fix' (self: rattrs self // {
```

Comment:
Please inline `fix'` here.
That will be at least as clear, while offsetting some of the minor performance loss in this PR.

Reply:
Oh yeah, I overlooked that in your earlier comment. I agree that `let` is clearer. Will fix when I get home

Reply:
Should we keep the `__unfix__` attribute added by `fix'`?

Reply:
I don't know. It seems like `__unfix__` is only used with `makeOverridable`; not `makeExtensible*`. I may be reading it wrong though.

I suppose we could ask users in a deprecation message, but maybe that's for a follow-up PR.

Reply:
Judging from the initial discussion when this was introduced (https://github.com/NixOS/nixpkgs/pull/15095#issuecomment-216246576, https://github.com/NixOS/nixpkgs/pull/15095#discussion_r82450528), I'd say `__unfix__` can be safely removed as its role is fulfilled by the extender attribute. I've left it in for now.